src/wrappers/hapi/HapiWrapper.ts

```diff
@@ -0,0 +1,195 @@
+import ModuleUtils from '../../utils/ModuleUtils';
+
+import { ApplicationManager } from '../../application/ApplicationManager';
+import Reporter from '../../Reporter';
+import ConfigProvider from '../../config/ConfigProvider';
+import ExecutionContextManager from '../../context/ExecutionContextManager';
+import ExecutionContext from '../../context/ExecutionContext';
+
+import ThundraLogger from '../../ThundraLogger';
+import { ClassNames, DomainNames } from '../../Constants';
+import LambdaUtils from '../../utils/LambdaUtils';
+import WrapperUtils from '../WebWrapperUtils';
+
+import PluginContext from '../../plugins/PluginContext';
+import ThundraConfig from '../../plugins/config/ThundraConfig';
+
+import * as HapiExecutor from './HapiExecutor';
+
+/**
+ * Create reporter instance
+ * @param {string} apiKey
+ * @return {Reporter} reporter
+ */
+function createReporter(apiKey: string): Reporter {
+    return new Reporter(apiKey);
+}
+
+/**
+ * Create plugin context instance
+ * @param {apiKey} apiKey
+ * @return {PluginContext} pluging context
+ */
+function createPluginContext(apiKey: string): PluginContext {
+    return WrapperUtils.createPluginContext(apiKey, HapiExecutor);
+}
+
+/**
+ * Handle Hapi server creation process
+ * @param {ThundraConfig} config
+ * @param {pluginContext} PluginContext
+ * @return {any[]} plugings
+ */
+function createPlugins(config: ThundraConfig, pluginContext: PluginContext): any[] {
+    return WrapperUtils.createPlugins(config, pluginContext);
+}
+
+/**
+ * Handle Hapi server creation process
+ * @param {Function} wrappedFunction
+ */
+function hapiServerWrapper(wrappedFunction: Function) {
+
+    ThundraLogger.debug('<HapiWrapper> Creating Thundra middleware ...');
+
+    ApplicationManager.setApplicationInfoProvider().update({
```

Comment:
This part

```js
ApplicationManager.setApplicationInfoProvider().update({
        applicationClassName: ClassNames.HAPI,
        applicationDomainName: DomainNames.API,
    });

    const appInfo = ApplicationManager.getApplicationInfo();
    ApplicationManager.getApplicationInfoProvider().update({
        applicationId: WrapperUtils.getDefaultApplicationId(appInfo),
    });

    const config = ConfigProvider.thundraConfig;
    const { apiKey } = config;

    const pluginContext = createPluginContext(apiKey);
    const reporter = __PRIVETE__.createReporter(apiKey);
    const plugins = createPlugins(config, pluginContext);
```

is running multiple times (creates different instances of plugins and reporter) because `hapiServerWrapper` is called multiple times while patching. So it is better to move this logic into init phase to run only once

Reply:
Also, is it possible to move this logic into `WebWrapperUtils`, take `applicationClassName` as param and call it from there and then return reporter and plugins here